src/metabase/db/custom_migrations/metrics_v2.clj

```diff
@@ -0,0 +1,158 @@
+(ns metabase.db.custom-migrations.metrics-v2
+  (:require
+   [cheshire.core :as json]
+   [metabase.util :as u]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2])
+  (:import (java.time Instant)))
+
+(set! *warn-on-reflection* true)
+
+(defn- get-metric-migration-collection
+  "Get the collection used to store the questions created from the v1 metrics.
+  If and only if the collection doesn't exists and the option `create?` is truthy,
+  the collection is created."
+  ([]
+   (get-metric-migration-collection nil))
+  ([{:keys [create?]}]
+   (let [coll-name "Migrated Metrics v1"
+         slug (u/slugify coll-name 510)
+         desc "This collection and the metrics in it were automatically generated by the migration to v50."]
+     (or (t2/select-one :collection :name coll-name :slug slug :description desc)
+         (when create?
+           (let [collection (t2/insert-returning-instance!
+                             :collection
+                             {:name coll-name, :slug slug, :description desc})]
+             (let [all-users-group-id (t2/select-one-fn :id :permissions_group :name "All Users")]
```

Comment:
I don't really love how we have the string literal `"All Users"` copied all over the codebase versus making it a real constant somewhere but I guess that's a problem for another day

Reply:
There is a var for that, I just didn't want to refer to it from migration code.

Reply:
Fair enough